### PR TITLE
Add rollback functionality to setup wizard

### LINKED ISSUE #9: if an error occurs while using the setup wizard, undo any actions that have succeeded

### PR DESCRIPTION
**Pull request by HYPRGK**
Fixes #9

Add error handling and rollback functionality to the setup wizard.

* Add `rollback_actions` function in `agentstack/cli/cli.py` to undo actions like directory creation and file writing.
* Update `init_project_builder` in `agentstack/cli/cli.py` to catch exceptions and call `rollback_actions`.
* Track created files and directories in `insert_template` for potential rollback in `agentstack/cli/cli.py`.
* Add `rollback_actions` function in `agentstack/generation/agent_generation.py` and `agentstack/generation/task_generation.py` to undo actions.
* Update `generate_agent` in `agentstack/generation/agent_generation.py` and `generate_task` in `agentstack/generation/task_generation.py` to catch exceptions and call `rollback_actions`.
* Add tests in `tests/test_cli_loads.py` to verify rollback functionality.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/AgentOps-AI/AgentStack/pull/75?shareId=663e8c7c-00c6-4985-a60e-61a8d9259537).

### CODE REVIEW

File: agentstack/cli/cli.py [hunk 1]
```diff
@@ -20,86 +20,100 @@
...
-    if slug_name and not is_snake_case(slug_name):
-        print(term_color("Project name must be snake case", 'red'))
-        return
+def rollback_actions():
+    for file in created_files:
+        if os.path.exists(file):
+            os.remove(file)
+    for dir in created_dirs:
+        if os.path.exists(dir):
+            shutil.rmtree(dir)
```
Comment (tcdent) [inline agentstack/cli/cli.py:32]: We should be very careful about deleting files when any exception occurs. Possible that running this command on an already initialized project would permanently erase user data.

**Comment by tcdent** [comment]
Thanks for taking the time to submit this PR!

I like the sentiment here, but we need to be careful about the implementation. 

The module-level variable names and rollback functions have no indication of what feature they're specifically related to. 

Stylistically, indenting the entire method inside a `try/except` statement is a bit much. 

Potential solution to this is to create a `forward` and `backward` method inside the relevant function which makes the scope more implicit. 

```python
def init_project_builder(...):
    def forward():
        ...
    def backward():
        ...

    try:
        forward()
    except:
        backward()
```

**Comment by bboynton97** [comment]
closing as stale. i'd love to have your contribution later on though :)

**Comment by potpiedev** [comment]
Starting Review, should be available shortly

**Pull request closed without merging**